src/meta-srv/src/bootstrap.rs

```diff
@@ -110,12 +110,14 @@ impl MetaSrvInstance {
         let addr = self.opts.http.addr.parse().context(error::ParseAddrSnafu {
             addr: &self.opts.http.addr,
         })?;
-        let http_srv = self.http_srv.start(addr);
-        select! {
-            v = meta_srv => v?,
-            v = http_srv => v.map(|_| ()).context(error::StartHttpSnafu)?,
-        }
-
+        let http_srv = async {
+            self.http_srv
+                .start(addr)
+                .await
+                .map(|_| ())
+                .context(error::StartHttpSnafu)
+        };
+        future::try_join(meta_srv, http_srv).await?;
```

Comment:
This seems a seprated fix alongside this refactor?

Reply:
Kind of. With http server becomes non-blocking, we have to change the codes here.